### DISCUSSION TITLE
Configuration for using existing reverse proxy

### DISCUSSION THREAD

**Discussion by arpitgupta** [teslamate-org/teslamate#1870] [Q&A]
So i see the instructions here for [traefik](traefik). What i am trying to figure out is what configs are needed if i already have a reverse proxy running? I am running a swag container which is managing the ssl cert and various reverse proxy configurations. Do i need to set the `VIRTUAL_HOST`, `CHECK_ORIGIN` environment variables or just adding a new reverse proxy configuration to my swag container be enough? From the traefik configs looks like the proxy needs to handle these paths?

```
- "traefik.http.routers.grafana.rule=Path(`/grafana`) || PathPrefix(`/grafana/`)"
- "traefik.http.routers.teslamate-ws.rule=Host(`${FQDN_TM}`) && Path(`/live/websocket`)"
         
```

**Comment by instantdreams** [comment]
I'm using Nginx Proxy Manager and all I need is to adjust the Grafana values to incldue a subpath. Looking into how to do this.

**Reply by LukeSkywalker993** [reply]
Do u have a solution?

**Reply by instantdreams** [reply]
Yes, I added this to the grafana section of my docker-compose.yml:

`      - GF_SERVER_DOMAIN=${DOMAIN}`

I made sure my .env file had an entry for DOMAIN that matched my subdomain.

**Reply by mikeh2010** [reply]
@instantdreams Hey great! that worked for grafana.

But only if I go directly to Grafana - I still have the issue with teslamate though...

**Reply by instantdreams** [reply]
I'm never exposing Teslamate to the internet, as there is no password protection.

**Reply by mikeh2010** [reply]
Ah OK.
I used this guide.  https://docs.teslamate.org/docs/guides/apache#docker-composeyml for password protection.

**Reply by LukeSkywalker993** [reply]
> - GF_SERVER_DOMAIN=${DOMAIN}

i added this, use "docker-compose up" and restartet the container.

but it's not working :-(
What is wrong?

and what have i todo with the env? I use nginx, there is no env.


![grafik](https://user-images.githubusercontent.com/59370744/175778093-a506d48b-7e70-4c5c-83b1-c1e9c6331d1e.png)

![grafik](https://user-images.githubusercontent.com/59370744/175778184-52fee1f7-fcae-41a5-85aa-c55eef6e64b5.png)
![grafik](https://user-images.githubusercontent.com/59370744/175778239-30466f70-a0db-493a-bec4-8e94b05dd33d.png)

@instantdreams i use the password protection from nginx.
